src/vm/threads.cpp

```diff
@@ -9058,9 +9061,15 @@ void Thread::DeleteThreadStaticData()
     // Deallocate the memory used by the table of ThreadLocalBlocks
     if (m_pThreadLocalBlock != NULL)
     {
-        m_pThreadLocalBlock->FreeTable();
-        delete m_pThreadLocalBlock;
-        m_pThreadLocalBlock = NULL;
+        PTR_ThreadLocalBlock pThreadLocalBlock = m_pThreadLocalBlock;
+
+        {
```

Comment:
What is keeping the `Thread` alive while this is running - can the `Thread` be released while this is running?

Reply:
The Thread::DeleteThreadStaticData() is called on the thread itself. And the Thread::Thread::DeleteThreadStaticData(ModuleIndex index) is called under the thread store lock here:
https://github.com/dotnet/coreclr/blob/master/src/vm/ceeload.cpp#L2910-L2915